test/cbmc/proofs/UDP/vProcessGeneratedUDPPacket_IPv6/vProcessGeneratedUDPPacket_IPv6_harness.c

```diff
@@ -0,0 +1,131 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+#include "list.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_ND.h"
+#include "FreeRTOS_UDP_IP.h"
+#include "FreeRTOS_IP_Private.h"
+
+/* Include the stubs for APIs. */
+#include "memory_assignments.c"
+#include "freertos_api.c"
+
+/* We do not need to calculate the actual checksum for the proof to be complete.
+ * Neither does the checksum matter for completeness. */
+uint16_t usGenerateChecksum( uint16_t usSum,
+                             const uint8_t * pucNextData,
+                             size_t uxByteCount )
+{
+    __CPROVER_assert( pucNextData != NULL, "Next data in GenerateChecksum cannot be NULL" );
+
+    uint16_t usChecksum;
+
+    /* Return any random value of checksum since it does not matter for CBMC checks. */
+    return usChecksum;
+}
+
+
+/* We do not need to calculate the actual checksum for the proof to be complete.
+ * Neither does the checksum matter for completeness. */
+uint16_t usGenerateProtocolChecksum( const uint8_t * const pucEthernetBuffer,
+                                     size_t uxBufferLength,
+                                     BaseType_t xOutgoingPacket )
+{
+    __CPROVER_assert( pucEthernetBuffer != NULL, "The Ethernet buffer cannot be NULL while generating Protocol Checksum" );
```

Comment:
Should we assert on the ethernet header being non-NULL as well?

Reply:
I see that assert is there in the usGenerateProtocolChecksum as well. So for generalization if there is assert in main function we should do __CPROVER_assert for that?